### ISSUE TITLE
Unexpected behaviour when new size is larger than full image

### ISSUE THREAD

**Issue by smerriman** [Automattic/regenerate-thumbnails#86]
Suppose a site has post thumbnails set to 200x200; I change this to 600x600 and regenerate thumbnails (telling the plugin to skip correctly sized images).

If after doing so, I upload a new 400x400 image, the post-thumbnail size won't get created in the metadata (since it is too small), so when the theme calls the_post_thumbnail(), it will output the full 400x400 image.

However, if a 400x400 image was uploaded prior to regeneration, when the theme calls the_post_thumbnail, it will output the original 200x200 image.

This is because the plugin is skipping the thumbnail size ('thumbnail would be larger than original'), but leaving the old size in the metadata, rather than removing it.

I believe removing it would be the logical behaviour.

**Comment by ddur** [comment]
AFAIK, WP 5.3 and above will clear all metadata[sizes] on upload and on regenerate

**Comment by smerriman** [comment]
No, this is nothing to do with the WP version (it would in fact be very bad if it cleared the metadata in general, since it's often important to have unregistered sizes).

**Comment by ddur** [comment]
https://github.com/WordPress/WordPress/blob/618329a7504958c58b2bd9898bdd8c49d65d386e/wp-admin/includes/image.php#L239

**Comment by smerriman** [comment]
Which is called when you upload a new image.. which doesn't seem to have any relevance to this plugin or thread.

See this section of code, which is important to retain (given the need to retain old metadata, eg things like WordPress' site icon which is an unregistered size), just needs tweaking based on the initial bug report:

https://github.com/Automattic/regenerate-thumbnails/blob/master/includes/class-regeneratethumbnails-regenerator.php#L199

**Comment by smerriman** [comment]
This is a bug report for the Regenerate Thumbnails plugin. It adds the sizes back in after calling the core WordPress function:

https://github.com/Automattic/regenerate-thumbnails/blob/master/includes/class-regeneratethumbnails-regenerator.php#L199

As mentioned, adding these sizes back is a vital part of the plugin - for example, at the point you upload a Site Icon to WordPress, it adds some square sizes to the metadata unique to that image so that it doesn't need to register a size and have it generated for your entire Media library. If these weren't readded, Site Icons would break after using this plugin.

So that section of the code shouldn't be removed; unregistered sizes should be re-added, but registered sizes - which may not exist in the new metadata due to the fact the original image isn't large enough - shouldn't.
